src/cli/repl/core.ts

```diff
@@ -21,33 +21,90 @@ const replCompleterKeywords = Array.from(commandNames, s => `:${s}`)
  * Used by the repl to provide automatic completions for a given (partial) input line
  */
 export function replCompleter(line: string): [string[], string] {
-	const singleCommand = replCompleterKeywords.find(k => (k + ' ') === line)
-	if(singleCommand !== undefined){
-		const applicableCommand = getCommand(singleCommand.substring(1))
-		guard(applicableCommand !== undefined, 'Command should be defined')
-		if(applicableCommand.script){
-			let scriptOptions: OptionDefinition[] | undefined
-			for(const [script, { options}] of Object.entries(scripts)){
-				if(script === line.trimEnd().substring(1)){
-					scriptOptions = options
-					break
-				}
-			}
-			guard(scriptOptions !== undefined, 'script should be in script record')
-			const autoCompleteList : string[] = []
-			autoCompleteList.push(line)
-			scriptOptions.forEach(option => {
-				autoCompleteList.push(line + ' --' + option.name)
-				if(option.alias !== undefined){
-					autoCompleteList.push(line + ' -' + option.alias)
+	const splitCommandList = splitAtEscapeSensitive(line)
+	console.log(splitCommandList)
+	//find command in commandlist
+	const keyword = replCompleterKeywords.find(k => splitCommandList[0] === k)
```

Comment:
For this check we are probably better off with a set, given that more and more commands are added.

Reply:
Is it possible for several keywords to be exactly equal?
In this instance it can only be one keyword. Shouldn't multiple commands never be mapped to one keyword

Reply:
my comment was about the speed - with find you do a linear search on all keywords. for the autocompletion to be faster, with more and more commands, i think a set would be fine because we only check if a given command exists.